source/slang/slang-ir.cpp

```diff
@@ -6426,7 +6452,14 @@ namespace Slang
         this->parent = inParent;
         
 #if _DEBUG
-        validateIRInstOperands(this);
+        // Disabling this for now. There's a case in 
+        // constructSSA that clones decorations on a param
+        // that hasn't yet been inserted into a block.
+        // This causes validation to fail because the decoration
+        // type is in IRModule, but the param is not nested under the 
+        // module yet.
+        // 
+        // validateIRInstOperands(this);
```

Comment:
This validation is here to make sure we don't introduce invalid IR during the autodiff process. Is there a way to keep it here? e.g. inserting the types referenced by the decoration first.

Reply:
If not possible, use `disableIRValidationAtInsert()` before inserting the violating inst and `enableIRValidationAtInsert()` after inserting.

Reply:
Removed this change, and surrounded the `constructSSA()` with `disableIRValidationAtInsert()`  and `enableIRValidationAtInsert()`